src/main/java/org/gooru/insights/services/ESDataProcessorServiceImpl.java

```diff
@@ -515,7 +521,9 @@ private void fetchMetrics(JSONObject newJson, List<Map<String, Object>> dataList
 				} else {
 					resultMap.put(entry.getKey(), aggregatedObject.get(APIConstants.FormulaFields.VALUE.getField()));
 				}
-				resultMap.put(groupBy[counter], newJson.get(APIConstants.FormulaFields.KEY.getField()));
+				if(groupBy != null && groupBy.length > 0) {
+					resultMap.put(groupBy[counter], newJson.get(APIConstants.FormulaFields.KEY.getField()));
```

Comment:
@RenukaGurumurthy  groupBy field is passed as null if it has only aggregation so null check is valid,but if it has groupBy then it sure has data so groupBy length > 0 is invalid validation.so remove it.